hs_core/hydroshare/utils.py

```diff
@@ -362,9 +362,9 @@ def replicate_resource_bag_to_user_zone(user, res_id):
     if istorage.exists(res_coll):
         bag_modified = istorage.getAVU(res_coll, 'bag_modified')
     if bag_modified == "true":
+        # import here to avoid circular import issue
+        from hs_core.tasks import create_bag_by_irods
         create_bag_by_irods(res_id, istorage)
```

Comment:
Instead of just directly calling it, you need to do something like `create_bag_by_irods.delay(res_id, istorage)` or `create_bag_by_irods.apply_async((res_id, istorage))` (yes, the tuple here is deliberate, see: http://celery.readthedocs.io/en/latest/userguide/calling.html#basics).  Calling it as if it is a plain function bypasses running it in a worker.